## Importar librerías

In [1]:
import pandas as pd
import numpy as np
import plotly.express as px

## Importar datos

In [2]:
df = pd.read_csv('Datasets/datos_limpios.csv')
pd.options.display.max_columns = None

In [3]:
def Analisisbivariado(variables,orient,mode,color,):
    contingency_table=pd.crosstab(df[variables[0]],df[variables[1]])
    contingency_table = contingency_table.div(contingency_table.sum(axis=1), axis=0)
    fig=px.bar(contingency_table,orientation=orient,barmode=mode,color_discrete_sequence=color)
    fig.update_layout(width=800,title=dict(text=f'{variables[0]} & {variables[1]}', x=0.5))
    fig.update_traces(texttemplate='%{value:.2%}', textposition='outside')
    print("Tabla de contingencia:")
    display(contingency_table)
    fig.show()


In [4]:
Analisisbivariado(['attrition','department'],'v','group',px.colors.qualitative.Plotly)

Tabla de contingencia:


department  Human Resources  Research & Development     Sales
attrition                                                    
No                 0.035753                0.656826  0.307421
Yes                0.078984                0.637518  0.283498

Se observa que el pertenecer al departamento de recursos humanos si podria afecta a la decision de renunciar ya que tenemos que de los que si renunciaron el 7.90% pertenecia a dicho departamento frente al 3.58% de los que no renunciaron, una diferencia de mas del doble, en los otros departamentos se ve una diferencia pero es menor

In [5]:
Analisisbivariado(['attrition','business_travel'],"h","group",px.colors.qualitative.T10)

Tabla de contingencia:


business_travel  Non_Travel  Travel_Frequently  Travel_Rarely
attrition                                                    
No                 0.111863           0.168743       0.719393
Yes                0.050776           0.290550       0.658674

La Frecuencia con la que los empleados estan viajando tambien podria estar afectando la decision de dejar la empresa ya que vemos una gran diferencia en la decision cuando se viaja frecuentemente y cuando no se viaja

In [6]:
Analisisbivariado(['attrition','education'],"v","relative",px.colors.qualitative.T10)

Tabla de contingencia:


education    Doctor  Licenciado    Máster  Por debajo de la universidad  \
attrition                                                                 
No         0.033315    0.391658  0.272210                      0.117010   
Yes        0.029619    0.373766  0.262341                      0.110014   

education  Universitario  
attrition                 
No              0.185807  
Yes             0.224260

El nivel de educacion de los empleados no parece afectar la decision ya que se distribuye mas o menos igual entre las dos opciones, hay unas ligeras diferencias en algunos niveles pero nada que parezca significativo

In [7]:
Analisisbivariado(['attrition','education_field'],"v","relative",px.colors.qualitative.Plotly)

Tabla de contingencia:


education_field  Human Resources  Life Sciences  Marketing   Medical  \
attrition                                                              
No                      0.013001       0.408992   0.108884  0.315547   
Yes                     0.046544       0.425952   0.105783  0.315938   

education_field     Other  Technical Degree  
attrition                                    
No               0.058505           0.09507  
Yes              0.042313           0.06347

A primera vista todo parece estar muy parejo entre ambas decisiones tal vez haya que fijarse en el campo de los recursos humanos y el del grado tecnico ya que presentan ciertas diferencias entre ellos pero los demas estan muy parejos

In [8]:
Analisisbivariado(['attrition','gender'],"h","group",px.colors.qualitative.Plotly)

Tabla de contingencia:


gender       Female      Male
attrition                    
No         0.404117  0.595883
Yes        0.379408  0.620592

El genero podemos decir que no afecta a la decision ya que se distribuye igual entre las opciones

In [10]:
Analisisbivariado(['attrition','job_role'],"v","relative",px.colors.qualitative.T10)

Tabla de contingencia:


job_role   Healthcare Representative  Human Resources  Laboratory Technician  \
attrition                                                                      
No                          0.091008         0.036566               0.175785   
Yes                         0.080395         0.029619               0.176305   

job_role    Manager  Manufacturing Director  Research Director  \
attrition                                                        
No         0.071506                0.104009           0.049567   
Yes        0.059238                0.067701           0.078984   

job_role   Research Scientist  Sales Executive  Sales Representative  
attrition                                                             
No                   0.194204         0.219935              0.057421  
Yes                  0.224260         0.232722              0.050776

El rol de trabajo tampoco parece estar afectando la decision, hay cierta diferencia cuando el rol es director de manufactura o director de investigacion pero nada demasiado significativo

In [11]:
Analisisbivariado(['attrition','marital_status'],"h","group",px.colors.qualitative.Plotly)

Tabla de contingencia:


marital_status  Divorced   Married    Single
attrition                                   
No              0.238082  0.477519  0.284399
Yes             0.139633  0.354020  0.506347

El estado civil de los trabajadores si parece estar afectando claramente la decision, si la persona esta soltera es mucho mas probable que se vaya a que se quede, si esta casada es un poco mas probable que no renuncie

In [12]:
Analisisbivariado(['attrition','environment_satisfaction'],"v","relative",px.colors.qualitative.T10)

Tabla de contingencia:


environment_satisfaction      Alto      Bajo     Medio  Muy alto
attrition                                                       
No                        0.320152  0.170910  0.196641  0.312297
Yes                       0.267983  0.300423  0.179126  0.252468

Si nos fijamos en la satisfacción con el ambiente de trabajo podemos notar que hay ciertas diferencias entre los niveles, particularmente cuando el nivel es bajo la probabilidad de que las personas se abandonen es casi del doble por lo que si podria estar afectando la decision

In [13]:
Analisisbivariado(['attrition','job_satisfaction'],"v","relative",px.colors.qualitative.Plotly)

Tabla de contingencia:


job_satisfaction      Alto      Bajo    Medio  Muy alto
attrition                                              
No                0.297941  0.179577  0.18987  0.332611
Yes               0.307475  0.277856  0.19323  0.221439

Analizando la satisfacción laboral podemos decir que las personas que sentian una satisfaccion alta se quedaron mucho mas mientras que los que sentian una baja satisfaccion tendieron a abandonar la empresa 

In [14]:
Analisisbivariado(['attrition','work_life_balance'],"h","group",px.colors.qualitative.T10)

Tabla de contingencia:


work_life_balance     Buena  La mejor      Mala  Muy buena
attrition                                                 
No                 0.229686  0.101029  0.044420   0.624865
Yes                0.241185  0.114245  0.105783   0.538787

El balance entre hogar y trabajo tambien podria estar afectando la decision ya que vemos que cuando hay un mal balance la probabilidad de abandonar es el doble a quedarse y cuando el balance es muy bueno la tendencia es ligeramente a quedarse.

In [15]:
Analisisbivariado(['attrition','job_involvement'],"h","group",px.colors.qualitative.T10)

Tabla de contingencia:


job_involvement      Alto      Bajo     Medio  Muy alto
attrition                                              
No               0.596425  0.052275  0.255688  0.095612
Yes              0.559944  0.076164  0.253879  0.110014

El nivel de implicación con el trabajo no parece afectar ya que es muy parecido entre ambas decisiones

In [20]:
Analisisbivariado(['attrition','performance_rating'],"v","relative",px.colors.qualitative.T10)

Tabla de contingencia:


performance_rating  Excelente  Sobresaliente
attrition                                   
No                   0.850217       0.149783
Yes                  0.826516       0.173484

El rendimiento de los trabajadores es una de las variables menos dicientes ya que todos los trabajadores fueron calificados de buena manera, habria que analizar porque sucede esto